cs-fun-practice-1.rb

```diff
@@ -0,0 +1,203 @@
+# `list = [3, 19, 27, 4, 98, 304, -9, 72]`
+
+# Then the call of `print(list)` should produce the following output:
+# `[3, 19, 27, 4, 98, 304, -9, 72]`
+
+# Your method should produce a single line of output (may wrap with long lists).
+
+def print_array(array)
+	# list = []
+
+	# list[0] = "["
+	# list << array
+	# list << "]"
+	print array
+	puts
+	# puts "...... there"
+end
+
+# ## Problem #2
+# Write a method named `stretch` that accepts an array of integers as a
+# parameter and returns a **new** array twice as large as the original, where
+# every integer from the original array is replaced with a pair of integers,
+# each half the original.
+
+# If a number in the original array is odd, then the first number in the new
+# pair should be one higher than the second so that the sum equals the
+# original number.
+
+# For example, if a variable named list refers to an array storing the
+# values [`18, 7, 4, 24, 11]`, the call of `stretch(list)` should return
+# a new array containing `[9, 9, 4, 3, 2, 2, 12, 12, 6, 5]`. (The number 18
+# is stretched into the pair 9, 9, the number 7 is stretched into 4, 3,
+# the number 4 is stretched into 2, 2, the number 24 is stretched into 12,
+# 12 and the number 11 is stretched into 6, 5.)
+
+def stretch(array)
+	new_array = []
+
+	first_array_length = array.length
+
+	num = 0
+
+	until new_array.length == (2 * first_array_length)
+		base_num = array[num]
+
+		if base_num % 2 == 0
+			even_num = base_num/2
+			
+			2.times do
+			new_array << even_num
+			end	
+		else
+			even_num = (base_num - 1) /2
+			odd_num = even_num + 1
+			new_array << odd_num
+			new_array << even_num
+		end
+		
+		num += 1
+	end
+	print new_array
+	puts
+
+end
+
+
+
+# ## Problem #3
+# Write a method named `numUnique` that accepts a sorted array of integers
+# as a parameter and **utilizes a hash to** calculate and return the number of
+# unique values in the array. The array is guaranteed to be in sorted order,
+# which means that duplicates will be grouped together.
+
+# For example, if a variable called list stores the following values:
+# `list = [5, 7, 7, 7, 8, 22, 22, 23, 31, 35, 35, 40, 40, 40, 41]`
+# then the call of `numUnique(list)` should return **9**
+# because this list has 9 unique values (5, 7, 8, 22, 23, 31, 35, 40 and 41).
+
+# It is possible that the list might not have any duplicates. For example if
+# list instead stored this sequence of values:
+# `list = [1, 2, 11, 17, 19, 20, 23, 24, 25, 26, 31, 34, 37, 40, 41]`
+# Then a call on the method would return **15**
+# because this list contains 15 different values.
+
+# If passed an empty list, your method should return **0**.
+
+
+
+def numUnique(array)
+	hash = {}
+
+	if array.length == 0
+		return "**0**"
+	else
+
+		num = 0
+
+		array.length.times do
+			next_num = (num + 1)
+
+			unless array[num] == array[next_num]
+				hash[num] = array[num] 
+			end
+
+			num += 1
+		end
+	end
+	return "**" + hash.length.to_s + "**"
+
+end
```

Comment:
We went over this one in class. 

Only additional feedback here is that you should be returning plain fixnums, not numbers buried in a String - if done the way I suggest, then you can utilize that information later if you wanted.